ideas/293-ulc-integration/README.md

```diff
@@ -0,0 +1,101 @@
+---
+id: 293-ulc-integration
+title: ULC integration
+status: draft
+created: 2018-08-06
+category: core
+lead-contributor: b00ris
+contributors:
+    - b00ris Go dev
+    - jeka Go dev
+exit-criteria: no
+success-metrics: no
+clear-roles: no
+future-iteration: no
+roles-needed:
+    - Go dev
+    - Go dev
+    - Clojure dev
+    - QA
+okrs:
+    - Core. [P0] LES and/or ULC are operational and used by at least 10% of all users.
+    - Research. Integrate and test ULC in a manner “graceful downgrade”: use infura by default and if it fails, start ULC node.
+---
+
+## Preamble
+
+    Idea: 293-ulc-integration
+    Title: ULC integration
+    Status: Draft
+    Created: 2018-08-06
+    Requires (*optional): #254
+
+## Summary
+Status app can use LES and ULC as Ethereum providers instead of Infura.
+
+## Swarm Participants
+- Lead Contributor: @b00ris
+- Contributor: @jeka
+- Contributor: @mandrigin
+- Contributor: TBD
+- QA: TBD
+
+## Product Overview
+We need LES client enabled on mobile device to be really decentralized and to enable all web3 features for dApps.
+
+### Requirements & Dependencies
+Depends on research swarm #254
+
+### Security and Privacy Implications
+* Free slots for LES servers
+* Trusted nodes selection
+
+### Minimum Viable Product
+* LES can work as the second Ethereum provider in status app.
+* ULC can work as the third Ethereum provider in status app with static trusted nodes.
+
+Goal Date: 2018-10-08
+
+Description:
+
+### Iteration 2018-08-13 - 2018-08-27
+
+Goal Date: 2018-08-27
+
+Description:
+* investigate les integration problems and fix them
```

Comment:
Perhaps we can be more precise here. I think it'd be useful to bring it back under a development flag as soon as possible, and figure out bugs as they are discovered. Doesn't seem realistic that all bugs will be fixed straight away, especially when they aren't clearly defined.

Reply:
Agree. I would put it as "return LES as a WIP option for testnets" as iteration 1 (I think that would be safer) and "make a list of issues of current LES implementation".
After this is done, we can really track progress and plan for that.
All the LES peers can be hardcoded for that, so we are putting discovery aside.